CodeforAustralia/council-emissions-calculator-spike
# Logging to LogFlare

## Purpose of this PR
To add some logging to capture user interactions with survey. This will give us some ability to troubleshoot issues on the frontend. See this [trello card](https://trello.com/c/jTbA7Zto)

This is my 2nd attempt at sending logs to LogFlare; first attempt in [this PR](https://github.com/CodeforAustralia/council-emissions-calculator-spike/pull/58).

### Notable changes/additions
- `pages/api/logger.js`: `logger` api to receive logs of client interactions, to send to LogFlare
- `utils/sendLogs.js`: helper function used in client end to send log
  messages to `logger` api route

N.B:
- the logging api route was implemented, so the api keys for LogFlare does not need to be exposed on the client side.
- logs sent to LogFlare using `POST` request with `fetch`
- the logging does not capture free text responses from Question 6 (question on incentives), to mitigate the likelihood of leaking out any PII info.
- the API key and source token for LogFlare is stored as an environment variable in Vercel (`LOGFLARE_API_KEY`, `LOGFLARE_SOURCE_TOKEN`)

### How to verify logging is working
1. go to [this link](https://logflare.app/sources/public/pZ0g2pCfpfDSoCaJ) to see the logs
2. fill out the survey on the [preview site](https://council-emissions-calculator-spike-git-fork-20bda0-civic-makers.vercel.app). Verify that a log entry appears When a navigation button (Start, Next, Back, Submit) is clicked. The entry should look something like this:
```
page: /form/Question6 | event: Back button clicked | km: 26 | nWorkDays: 3 | travelDays: | workMode: full work from home | mainTransportMode: Train/tram | incentive: <filled> | department: Parks & Open Spaces
```

### Refs:
  - https://logflare.app/guides/vercel-setup#browser-client-logs
  - https://github.com/Logflare/pino-logflare/blob/master/docs/VERCEL.md
